tfsdk/value_as_test.go

```diff
@@ -40,11 +89,225 @@ func TestValueAs(t *testing.T) {
 	}
 
 	tests := map[string]testCase{
-		"primitive": {
+		"primitive bool pointer": {
+			val:      types.Bool{Value: true},
+			target:   newBoolPointer(false),
+			expected: newBoolPointer(true),
+		},
+		"primitive bool pointer pointer": {
+			val:      types.Bool{Value: true},
+			target:   newBoolPointerPointer(false),
+			expected: newBoolPointerPointer(true),
+		},
+		"primitive float64 pointer": {
+			val:      types.Float64{Value: 12.3},
+			target:   newFloatPointer(0.0),
+			expected: newFloatPointer(12.3),
+		},
+		"primitive float64 pointer pointer": {
+			val:      types.Float64{Value: 12.3},
+			target:   newFloatPointerPointer(0.0),
+			expected: newFloatPointerPointer(12.3),
+		},
+		"primitive int64 pointer": {
+			val:      types.Int64{Value: 12},
+			target:   newInt64Pointer(0),
+			expected: newInt64Pointer(12),
+		},
+		"primitive int64 pointer pointer": {
+			val:      types.Int64{Value: 12},
+			target:   newInt64PointerPointer(0),
+			expected: newInt64PointerPointer(12),
+		},
+		// Following test fails as target.Type() is big.Float not *bigFloat.
```

Comment:
@bflad is this a bug? Just wondering if `into.go` needs to be amended to handle this use-case?

Reply:
My limited understanding with the [`math/big` package](https://pkg.go.dev/math/big) in general is to conventionally always use pointer types to save on various memory allocations and prevent issues with zero values on the struct. So doing something like:

```go
var myBigOleFloat big.Float
tfsdk.ValueAs(ctx, example, &myBigOleFloat)
```

Would be considered something of an anti-pattern. Could support it? I guess if it is not hard to implement it. _Should_ support it? Probably more debatable, although if the error behavior/messaging is unclear then maybe that adds some weight to supporting it.

Reply:
Ok. I'll leave this for now and re-visit if it becomes obvious that behaviour/messaging is unclear.